plutus-tx/src/PlutusTx/Show.hs

```diff
@@ -0,0 +1,63 @@
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE LambdaCase        #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# OPTIONS_GHC -Wno-orphans #-}
+
+module PlutusTx.Show (
+    Show (..),
+) where
+
+import PlutusTx.Base
+import PlutusTx.Bool
+import PlutusTx.Builtins
+import PlutusTx.Builtins.Internal qualified
+import PlutusTx.Foldable (foldl)
+import PlutusTx.Show.TH (Show (..), ShowS, showString)
+
+instance Show Integer where
+    {-# INLINEABLE show #-}
+    show = displayConstant
+
+instance Show BuiltinByteString where
+    {-# INLINEABLE show #-}
+    show x = displayConstant (fromBuiltin x)
+
+instance Show BuiltinData where
+    {-# INLINEABLE show #-}
+    show (PlutusTx.Builtins.Internal.BuiltinData x) = displayConstant x
+
+instance Show BuiltinString where
+    {-# INLINEABLE show #-}
+    show x = displayConstant (fromBuiltin x)
+
+instance Show Bool where
+    {-# INLINEABLE show #-}
+    show x = displayConstant x
+
+instance Show () where
+    {-# INLINEABLE show #-}
+    show () = displayConstant ()
+
+-- It is possible to make it so that when `a` is a builtin type, `show (xs :: [a])`
+-- is compiled into a single `displayConstant` call, rathern than `length xs` calls.
+-- To do so the plugin would need to try to solve the `PrettyConst [a]` constraint,
+-- and branch based on whether it is solvable. But the complexity doesn't seem to
+-- be worth it: the saving in budget is likely small, and on mainnet the trace messages
+-- are often erased anyway.
+--
+-- Same for the `Show (a, b)` instance.
+instance Show a => Show [a] where
```

Comment:
Why not be consistent with `base` and add `showsList`?

Reply:
I think base's `showsList` is specifically for `String` (otherwise why not add `showsPair`, `showsMaybe` etc.). We don't need `Show String` for our `Show`.

Reply:
Add this comment to the code please.